src/cdk/overlay/overlay-directives.ts

```diff
@@ -311,7 +312,12 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
     });
 
     this._overlayRef.outsidePointerEvents().subscribe((event: MouseEvent) => {
-      this.overlayOutsideClick.next(event);
+      const origin = this._getOriginElement();
+      const target = _getEventTarget(event) as Element | null;
+
+      if (!target || !origin || (origin !== target && !origin.contains(target))) {
```

Comment:
nit: the null checks appear to be redundant since `contains` returns false if `target` is null. This could be reduced to `origin !== target && !origin?.contains(target))`

Reply:
I'll get rid of the `target` check, but I'll leave in the one for `origin` since I don't want to end up comparing nulls.